### DISCUSSION TITLE
Heads up for v3.7.0

### DISCUSSION THREAD

**Discussion by janiversen** [pymodbus-dev/pymodbus#2123] [General]
We have started to add API changes to dev, as preparation for v3.7.0. In case we run into a bigger bug we can still make v3.6.7 without API changes.

The current plan from our side for v3.7.0 are:
- BaseClient changes inheritance of ModbusProtocol to self.transport
- integrate message decode for all types into the existing framer code, this will break code compatibility with sync, so the current code is kept and executed for sync clients. In essence execute() and async_execute() will use different decode calls.
- add a multidrop parameters for the serial server. Multidrop is already supported, but slows the normal server (1 client and 1 server on the rs485), the idea is to add a new "hunt" mode, to find the next frame for the server
- get test coverage over 91%, with message layer and below at 100%
- Add a "proof of concept" transaction layer, not integrated
- framer will only accept the new enum and not the old framer class
- your pull requests with changes you need (and which are useable for many).

Timeline is June/July.

Help is more than welcome, not just with code but just as much with documentation and test cases.

**Comment by MAKOMO** [comment]
Maybe missing on your list is the drop of Python 3.8 support which I spotted recently in one of your message. Maybe this does not hold any longer?

**Reply by janiversen** [reply]
Already dropped in commit 58a1c37d, but thanks for commenting.

Dropping support in reality only means removing it from CI, so we are no longer notified if new code breaks with 3.8.

Currently there are no specific 3.8 code (even though I still remember having an if 3.8 somewhere) but there are a couple of overlapping exceptions with OSError which were needed only for 3.8.

**Reply by janiversen** [reply]
Just to be clear that is on dev, not yet released.
